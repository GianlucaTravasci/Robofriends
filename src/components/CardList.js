import React from 'react';
import Card from './Card';

const CardList = ({robots}) => {
    /*if (true) { Checking the ErrorBoundary
        return new Error("nooo");
    }*/
    const robotsArray = robots.map((user, i) => {
        return (
            <Card
                key= {i}
                id = {robots[i].id}
                name = {robots[i].name}
                email = {robots[i].email}
            />
        );
    })
    return (
        <div>
            {robotsArray}
        </div>
    );
}

export default CardList;