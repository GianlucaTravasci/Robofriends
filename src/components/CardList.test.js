import React from 'react';
import { shallow } from 'enzyme';

import CardList from './CardList';

it('renders without crashing', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'john',
            username: 'JohnJohn',
            email: 'John@gmail.com'
        }
    ]
    const wrapper = shallow(<CardList robots={mockRobots} />)
    expect(wrapper.debug()).toMatchSnapshot();

});
