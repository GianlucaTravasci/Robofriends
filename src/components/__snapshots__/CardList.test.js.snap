// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
"<div>
  <Card id={1} name=\\"john\\" email=\\"John@gmail.com\\" />
</div>"
`;
